### PR TITLE
[#391] Implement update hooks for apigee_api_catalog 2.x update

### LINKED ISSUE #391: Document steps for upgrading to apigee_api_catalog 2.x
Related:

#363 and #390

### PR DESCRIPTION
**Pull request by shadcn**
Fixes #391 

Related: #363 and #390

### CODE REVIEW

**Review by arlina-espinoza — APPROVED** [review]
This looks good @arshad , thanks!
We just need to add a warning to the release docs about it the config that will get automatically updated, in case some users made some customizations already. Could you draft the release notes for this item @arshad ? Thanks!

**Comment by shadcn** [comment]
@arlina-espinoza I added some checks to only update the fields if they have not been changed.

(Adding the release notes below)

### Release notes

Note: If you're upgrading from Apigee API Catalog 1.x to 2.x, this Kickstart release includes update hooks for the `apidoc` form and view displays. It also replaces the `field_categories` filter in the API Catalog view with the new `field_categories_apidoc` field.

File: apigee_devportal_kickstart.install [hunk 1]
```diff
@@ -402,3 +403,154 @@ function apigee_devportal_kickstart_update_8007() {
...
+  $definitions = $entity_field_manager->getFieldDefinitions('node', 'apidoc');
+
+  /** @var \Drupal\Core\Entity\EntityDisplayRepository $entity_display_repository */
+  $entity_display_repository = \Drupal::service('entity_display.repository');
+  /** @var \Drupal\Core\Entity\Entity\EntityFormDisplay $entity_form_display */
+  $entity_form_display = $entity_display_repository->getFormDisplay('node', 'apidoc');
+  $entity_view_display = $entity_display_repository->getViewDisplay('node', 'apidoc', 'card');
+
+  $fields = [
+    'field_image_apidoc' => [
```
Comment (kscheirer) [inline apigee_devportal_kickstart.install:423]: The rest of the fields on this content type follow a format like `field_apidoc_XXX`, do you want to use that here? Just a minor note, thanks for the PR!

Comment (shadcn) [reply apigee_devportal_kickstart.install:423]: We can't really do this here because each field has distinct form/view display config. 👍

**Pull request merged**